### ISSUE TITLE
'Media Viewer+' - A Single Window for Viewing and Tagging Media Simultaneously

### ISSUE THREAD

**Issue by prof-m** [hydrusnetwork/hydrus#1337]
# Request
A new window that combines (most of) the functionality of Hydrus' current 'media viewer' window and 'manage tags' window into one single window.

# Highly Professional MS Paint Mockup
![hydrus-view-tag-one-window-mock](https://user-images.githubusercontent.com/65079055/219906932-3d9b0d0f-2213-474e-8cb8-6acfba61689f.jpg)

# Functionality outline
- New window contains two sections - one displaying the media, and one displaying the tags and tag functionality (shown above in a traditional left-right split)
- Tag section both shows current tags and allows the user to type in tags to add/remove them from the image being viewed. (Bonus points if we can make the display of "related/recent" optional? I never use those, but probably other folks do!)
- Ability to adjust how much width each section is taking up by clicking and dragging the line between the two sections.
- Nice to haves, but definitely not mvp:
  - Width auto-adjusting based on zoom preferences and image display size
  - Ability to set custom min-max width of each section

# Related Issues

- #718 
- #250 
- #984 

# Rationale

- Determining the right tags for an image frequently involves the act of looking at the image (at a decent resolution).
- Having to open (and often reposition) a new window to add/edit/remove tags from a single image is time-consuming
- When an image needs many tags added to it (say, a fresh, untagged image), a user may need to frequently look back and forth between the tags and the image to make sure it gets properly tagged. This could be a lot more easily done if the image were in the same window as the dialogue for managing tags!
- Less windows less problems

**Comment by NyaaTell** [comment]
Only as an optional alternative to the current solution.
Acceptable, if the tag editing popup pushes the media view aside into the remained or the window space - when it's called via a button or shurtcut.

Rationale - I don't want to see tags when I look at media.

**Comment by prof-m** [comment]
@NyaaTell Yes, definitely, thanks for the call out. I edited the feature request to make that explicit.

**Comment by prof-m** [comment]
Update for posterity:

Brought this up to Dev on Discord, and they really liked the idea. As usual, no exact timeline, but they said they hoped to include it whenever they start on a larger project to refactor the manage tags dialogue.

**Comment by Intcatma** [comment]
> Only as an optional alternative to the current solution. Acceptable, if the tag editing popup pushes the media view aside into the remained or the window space - when it's called via a button or shurtcut.
> 
> Rationale - I don't want to see tags when I look at media.

This, but have have the search and tag edit panels be tabs, just like how the preparation and filtering panels in the duplicates page are. The only problem is that the current tag editor already uses tabs, so there would be 3 rows of tabs in total which is too many.

**Comment by Intcatma** [comment]
I propose that we use tabs to select between the search and tag editing panels on the left, similar to the duplicates processing page. We should use something like a dropdown to select the tag service to edit, if we used tabs like it is in the current tag edit page there would be 3 tab bars which is too much.
